### PR TITLE
Fix segmented control golden tests

### PR DESCRIPTION
**Pull request by sandrasandeep**
Segmented control golden file tests are failing because the content renders differently on different types of machines. Currently, the golden files were generated on a Mac, so this PR adds code so that the tests only compare against the golden files if the tests are running on MacOS.

### CODE REVIEW

File: packages/flutter/test/cupertino/segmented_control_test.dart [hunk 1]
```diff
@@ -893,7 +919,13 @@ void main() {
       ),
     );
 
-    await expectLater(find.byType(RepaintBoundary), matchesGoldenFile('segmented_control_test.0.0.png'));
+    // if golden files are regenerated on a non-Mac machine,
+    // change the 'skip' line to only run this test on that type of machine
```
Comment (amirh) [inline packages/flutter/test/cupertino/segmented_control_test.dart]: Explain why we are only running this on Mac?

File: packages/flutter/test/cupertino/segmented_control_test.dart [hunk 2]
```diff
@@ -893,7 +919,13 @@ void main() {
...
+    await expectLater(
+      find.byType(RepaintBoundary),
+      matchesGoldenFile('segmented_control_test.0.0.png'),
+      skip: !Platform.isMacOS,
```
Comment (amirh) [inline packages/flutter/test/cupertino/segmented_control_test.dart]: Do we want to skip the entire test on non Mac machines, or just not make assertions?

File: packages/flutter/test/cupertino/segmented_control_test.dart [hunk 3]
```diff
@@ -893,7 +919,13 @@ void main() {
       ),
     );
 
-    await expectLater(find.byType(RepaintBoundary), matchesGoldenFile('segmented_control_test.0.0.png'));
+    // if golden files are regenerated on a non-Mac machine,
```
Comment (amirh) [inline packages/flutter/test/cupertino/segmented_control_test.dart]: nit: start sentences with a capital letter, end with a period.

File: packages/flutter/test/cupertino/segmented_control_test.dart [hunk 4]
```diff
@@ -925,6 +957,12 @@ void main() {
     await tester.startGesture(center);
     await tester.pumpAndSettle();
 
-    expect(find.byType(RepaintBoundary), matchesGoldenFile('segmented_control_test.1.0.png'));
+    // if golden files are regenerated on a non-Mac machine,
```
Comment (amirh) [inline packages/flutter/test/cupertino/segmented_control_test.dart]: Same comment as above.

**Review by amirh — CHANGES_REQUESTED** [review]
Thanks Sandra!

A nit for future CL - Would make it easier to review if you separate formatting from actual changes.
Also try to add some more details in the PR description: what was the problem, how are we fixing it.

File: packages/flutter/test/cupertino/segmented_control_test.dart (hunk 3 above)
Comment (amirh) [reply packages/flutter/test/cupertino/segmented_control_test.dart]: Not like I did here :smile:

File: packages/flutter/test/cupertino/segmented_control_test.dart (hunk 1 above)
Comment (sandrasandeep) [reply packages/flutter/test/cupertino/segmented_control_test.dart]: I have added the explanation to the comment.

File: packages/flutter/test/cupertino/segmented_control_test.dart (hunk 2 above)
Comment (sandrasandeep) [reply packages/flutter/test/cupertino/segmented_control_test.dart]: I think we should just not make the assertion. The 'skip' line should accomplish this.

File: packages/flutter/test/cupertino/segmented_control_test.dart [hunk 5]
```diff
@@ -893,7 +919,16 @@ void main() {
       ),
     );
 
-    await expectLater(find.byType(RepaintBoundary), matchesGoldenFile('segmented_control_test.0.0.png'));
+    // If golden files are regenerated on a non-Mac machine,
```
Comment (amirh) [inline packages/flutter/test/cupertino/segmented_control_test.dart]: See if you can simplify this explanation.

**Review by amirh — APPROVED** [review]
LGTM

**Comment by mravn-google** [comment]
Landing this to get the tree back to green. The error to be fixed just appeared six times in a row.

Fixes https://github.com/flutter/flutter/issues/18737

**Comment by amirh** [comment]
This shouldn't have caused a red rope... the problem fixed here was a test that should have fail wasn't failing.
(so you see the "Exception caught by testing framework" message but the test doesn't actually fail).

**Comment by mravn-google** [comment]
Yeah, I misread [this log](https://ci.appveyor.com/project/flutter/flutter/build/1.0.14506) of a failing AppVeyor build. The segmented_control_test stacktrace was the only visible error in there.

**Comment by mravn-google** [comment]
Of course, whether or not tests fail, we shouldn't have stack traces in our test logs. But the real question is: did I land this before you were done with it?

**Comment by sandrasandeep** [comment]
Hi Mikkel, I thought I wasn't done with it, but it's fine that you merged it if all the tests were passing.

**Pull request merged**